src/common/snippets/include/snippets/pass_manager.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "openvino/pass/manager.hpp"
+#include "openvino/pass/pass.hpp"
+#include "openvino/pass/validate.hpp"
+#include <typeinfo>
+
+
+namespace ov {
+namespace snippets {
+namespace pass {
+/**
+ * @brief Manager is like ov::pass::Manager, but allows to insert new passes at arbitrary places in the pipeline
+ * @ingroup snippets
+ */
+class Manager : public ov::pass::Manager {
+public:
+    ~Manager() override = default;
+    using PassBase = ov::pass::PassBase;
+    using Validate = ov::pass::Validate;
+    /**
+    * @brief PassPosition describes a particular position in a transformation pipeline,
+     * where a new transformation should be inserted.
+     * @param pass_name name of the anchor pass, the new pass will be inserted before/after it.
+     * Empty pass_name could mean either beginning or the end of the pipeline depending on the `after` flag.
+     * No default value.
+     * @param after `true` if the new pass should be inserted before the anchor pass, `false` otherwise (default).
+     * If `pass_name` is empty, `true` means the end, and `false` - the beginning of the pipeline.
+     * @param pass_instance the number of the pass with matching `pass_name` to be considered as the anchor pass.
+     * 0 (default) means the first pass with `pass_name` will be considered as the anchor pass.
+    * @ingroup snippets
+    */
+    class PassPosition {
+        const std::string m_pass_name;
+        const size_t m_pass_instance;
+        const bool m_after;
+
+    public:
+        using pass_list_type = std::vector<std::shared_ptr<ov::pass::PassBase>>;
```

Comment:
If we use camel case for class names, we should use it for `using` as well 🙂 
```suggestion
        using PassListType = std::vector<std::shared_ptr<ov::pass::PassBase>>;
```

Reply:
Ok, done